src/main/webapp/placeGuides/PlaceGuideRepository.js

```diff
@@ -142,18 +146,23 @@ class PlaceGuideRepository {
 
   togglePlaceGuideBookmarkStatus(placeGuideId) {
     // Toggle in in-memory dictionary.
-    var isBookmarked = this._placeGuides.bookmarkedByCurrentUser;
-    this._placeGuides.bookmarkedByCurrentUser = !isBookmarked;
+    var isBookmarked = this._placeGuides[placeGuideId].bookmarkedByCurrentUser;
+    this._placeGuides[placeGuideId].bookmarkedByCurrentUser = !isBookmarked;
     // Toogle in database.
     var url = new URL("bookmark-place-guide", document.URL);
     url.searchParams.append("placeGuideId", placeGuideId);
-    if (this._placeGuides.bookmarkedByCurrentUser) {
-      url.searchParams.append("bookmarkHandlingType", "BOOKMARK");
-    } else {
+    if (isBookmarked) {
       url.searchParams.append("bookmarkHandlingType", "REMOVE");
+    } else {
+      url.searchParams.append("bookmarkHandlingType", "BOOKMARK");
     }
     return fetch(url)
-        .catch(error => console.log("BookmarkPlaceGuideServlet: failed to fetch: "
-            + error));
+        .catch(error =>
+            console.log("BookmarkPlaceGuideServlet: failed to fetch: "
```

Comment:
You should show an error message to the user as well (e.g. a toast) and in case of optimistic update revert the local cache.

Reply:
I added error messages and used the responsive server concept, so that I update the local cache only after the fetch was successfully executed.